packages/back-end/src/services/users.ts

```diff
@@ -89,3 +94,92 @@ export async function createUser(
     verified,
   });
 }
+
+/**
+ * Some tracking properties exist on the request object
+ * @param req
+ */
+export const getAuditableUserPropertiesFromRequest = (
+  req: Request
+): Pick<UserLoginAuditableProperties, "userAgent" | "device" | "ip" | "os"> => {
+  const userAgent = (req.headers["user-agent"] as string) || "";
+  const device = (req.headers["sec-ch-ua"] as string) || "";
+  const os = (req.headers["sec-ch-ua-platform"] as string) || "";
+  const ip =
+    (req.headers["x-forwarded-for"] as string) ||
```

Comment:
why not `req.ip`? Do our customers need the full list of proxied IPs, including those in our own cloud infra?

Reply:
I think `req.ip` will likely return the IP address of the nearest proxy to the server that delivered the request if I'm not mistaken

You bring up a good point about the full list  though - @tinahollygb this value can be a comma separated list of multiple IPs. If we only want the client IP, we'll want to split on `,` and take the first (leftmost) one

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For#syntax

Reply:
Good questions everybody. It looks like `req.ip` returns the leftmost X-Forwarded-For by default: https://expressjs.com/en/api.html#req.ip

I can change it to that, or I can leave the raw header for us to be more flexible and client-side, split on the commas if necessary. Or we can choose to store an array of IP's. I don't really have strong opinions about which approach we take. 

It seems like these are more so wonderings rather than actionable feedback. Do either @bryce-fitzsimons or @bttf have a strong preference for the approach we take here?

Reply:
Probably req.ip, as it already scrapes X-Forwarded-For when applicable, and is what we already use for generating refresh tokens.

Reply:
I've made the change to use `req.ip` in 3785aa18c5c5d3d827bc994a1360867180cd1380